src/actions/receive-profile.js

```diff
@@ -655,12 +653,34 @@ export function doneSymbolicating(): Action {
   return { type: 'DONE_SYMBOLICATING' };
 }
 
-export function coalescedFunctionsUpdate(
-  functionsUpdatePerThread: FunctionsUpdatePerThread
-): Action {
-  return {
-    type: 'COALESCED_FUNCTIONS_UPDATE',
-    functionsUpdatePerThread,
+export function coalescedSymbolicationStep(
+  symbolicationStepsPerThread: Map<ThreadIndex, SymbolicationStepInfo[]>
+): ThunkAction<void> {
+  return (dispatch, getState) => {
+    const { threads } = getProfile(getState());
+    const oldFuncToNewFuncMaps = new Map();
+    const symbolicatedThreads = threads.map((oldThread, threadIndex) => {
+      const symbolicationSteps = symbolicationStepsPerThread.get(threadIndex);
+      if (symbolicationSteps === undefined) {
+        return oldThread;
+      }
+      const oldFuncToNewFuncMap = new Map();
+      let thread = oldThread;
+      for (const symbolicationStep of symbolicationSteps) {
+        thread = applySymbolicationStep(
+          thread,
+          symbolicationStep,
+          oldFuncToNewFuncMap
+        );
+      }
+      oldFuncToNewFuncMaps.set(threadIndex, oldFuncToNewFuncMap);
+      return thread;
+    });
+    dispatch({
+      type: 'COALESCED_SYMBOLICATION_STEP',
+      oldFuncToNewFuncMaps,
```

Comment:
This could also be where we determine if functions were split, and then invalidate the state if that's the case.

Reply:
Hmm, I think `applySymbolicationStep` could produce a set of split funcs. Then we'd just forward that set here, and have the reducers invalidate the appropriate pieces of state.